assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.stories.js

```diff
@@ -153,6 +159,88 @@ MultipleRecoverableModuleNoAccess.scenario = {
 	delay: 250,
 };
 
+export const SingleRecoverableModuleError = Template.bind( {} );
+SingleRecoverableModuleError.storyName =
+	'Single Recoverable Module with Error Message';
+SingleRecoverableModuleError.args = {
+	setupRegistry: ( registry ) => {
+		const errorResponse = {
+			code: 'module_not_recoverable',
+			message: 'Module is not recoverable.',
+			data: { status: 403 },
+		};
+
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/modules\/data\/recover-module/,
+			{ body: errorResponse, status: 403 }
+		);
+
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveRecoverableModules( [ 'search-console' ] );
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveCheckModuleAccess(
+				{ access: true },
+				{ slug: 'search-console' }
+			);
+	},
+};
+SingleRecoverableModuleError.decorators = [
+	( Story ) => {
+		setTimeout( async () => {
+			await waitForElement( () => document.querySelector( 'button' ) );
+			document.querySelector( 'button' ).click();
+		}, 0 );
+
+		return <Story />;
+	},
+];
+
+export const MultipleRecoverableModuleErrors = Template.bind( {} );
+MultipleRecoverableModuleErrors.storyName =
+	'Multiple Recoverable Modules with Error Messages';
+MultipleRecoverableModuleErrors.args = {
+	setupRegistry: ( registry ) => {
+		const errorResponse = {
+			code: 'module_not_recoverable',
+			message: 'Module is not recoverable.',
+			data: { status: 403 },
+		};
+
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/modules\/data\/recover-module/,
+			{ body: errorResponse, status: 403 }
+		);
+
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveRecoverableModules( [ 'search-console', 'analytics' ] );
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveCheckModuleAccess(
+				{ access: true },
+				{ slug: 'search-console' }
+			);
+		registry
+			.dispatch( CORE_MODULES )
+			.receiveCheckModuleAccess(
+				{ access: true },
+				{ slug: 'analytics' }
+			);
+	},
+};
+MultipleRecoverableModuleErrors.decorators = [
+	( Story ) => {
+		setTimeout( async () => {
+			await waitForElement( () => document.querySelector( 'button' ) );
+			document.querySelector( 'button' ).click();
+		}, 0 );
```

Comment:
This isn't something we've done in other stories AFAIK mostly because errors are stored in the datastore automatically by our fetch store infrastructure, which makes it much more straightforward to put a component in such a state.

Let's source the error(s) from the store instead rather than component state using the `getErrorForAction` selector which every store should have.

Reply:
I have addressed this so that the error can be sourced directly from the reducer. Thank you!